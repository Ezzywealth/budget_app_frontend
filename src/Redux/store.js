import { configureStore } from '@reduxjs/toolkit';
import budgetReducer from './BudgetSlice';
import userReducer from './UserSlice';

const store = configureStore({
	reducer: {
		budget: budgetReducer,
		user: userReducer,
	},
});

export default store;
