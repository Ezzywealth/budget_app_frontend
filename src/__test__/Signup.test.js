import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import SignUpComponent from '../Components/Authentication/Signup';
import { describe, it, expect } from '@jest/globals';

describe('Delete Cab', () => {
	it('renders sign up component', () => {
		const tree = TestRenderer.create(
			<Provider store={store}>
				<BrowserRouter>
					<SignUpComponent />
				</BrowserRouter>
			</Provider>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('renders Sign up form', () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<SignUpComponent />
				</BrowserRouter>
			</Provider>
		);

		const header = screen.getByText(/Enter your details to sign up/i);
		expect(header).toBeInTheDocument();
	});
	it('renders sign up component and whows the already have an account test', async () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<SignUpComponent />
				</BrowserRouter>
			</Provider>
		);
		const subHead = screen.getByText(/Already have an account?/i);
		expect(subHead).toBeInTheDocument();
	});
});
