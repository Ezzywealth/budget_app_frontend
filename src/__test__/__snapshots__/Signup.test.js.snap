// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Cab renders sign up component 1`] = `
<section
  className="h-screen p-4 pt-[80px] laptop:px-20 overflow-auto  flex justify-center items-center w-full   "
>
  <div
    className="w-full relative flex justify-center"
  >
    <div
      className="w-full  mt-10 py-8  bg-white rounded-lg shadow-lg  my-12 md:my-4  xl:p-8 overflow-auto "
    >
      <div
        className="p-4 tablet:p-6 space-y-2 md:space-y-4 sm:p-8"
      >
        <h1
          className="text-base flex justify-center mb-8 font-bold leading-tight tracking-tight text-my-primary md:text-2xl  capitalize"
        >
          Enter your details to sign up
        </h1>
        <form
          className="space-y-4 tablet:space-y-6  w-full md:w-[450px]"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="block mb-2 text-base font-semibold text-gray-600 dark:text-white"
            >
              Name
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:border-my-primary focus:border-2 focus:outline-none focus:ring-0 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              disabled={false}
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="E-mail"
              type="text"
            />
          </div>
          <div>
            <label
              className="block mb-2 text-base font-semibold text-gray-600 dark:text-white"
            >
              E-mail
            </label>
            <input
              className="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:border-my-primary focus:border-2 focus:outline-none focus:ring-0 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              disabled={false}
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="E-mail"
              type="email"
            />
          </div>
          <div>
            <label
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            >
              Password
            </label>
            <input
              className="bg-gray-50 border-2 border-gray-300  text-gray-900 sm:text-sm rounded-lg focus:border-my-primary focus:border-2 focus:outline-none focus:ring-0 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400  dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              disabled={false}
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="••••••••"
              type="password"
            />
          </div>
          <button
            className="w-full text-white bg-[#00d960]  focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg flex justify-center items-center px-5 py-3 tracking-wider text-xl text-center  transition-all duration-300 ease-linear dark:hover:bg-primary-700 dark:focus:ring-primary-800"
            disabled={false}
            type="submit"
          >
            <span>
               Sign up
            </span>
          </button>
          <div
            className="flex flex-col"
          >
            <p
              className="text-sm font-light  text-gray-500 dark:text-gray-400"
            >
              Already have an account?
              <a
                className=" ml-2 hover:text-blue-500 font-semibold text-primary-600 hover:underline dark:text-primary-500"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>
</section>
`;
